WalletWasabi.Tests/RegTests.cs

```diff
@@ -1219,71 +1246,60 @@ public async Task BuildTransactionValidationsTestAsync()
 			var chaumianClient = new CcjClient(synchronizer, rpc.Network, keyManager, new Uri(RegTestFixture.BackendEndPoint), null);
 
 			// 6. Create wallet service.
-			var workDir = Path.Combine(Global.Instance.DataDir, nameof(SendTestsFromHiddenWalletAsync));
+			var workDir = Path.Combine(Global.Instance.DataDir, nameof(BuildTransactionValidationsTestAsync));
 			var wallet = new WalletService(bitcoinStore, keyManager, synchronizer, chaumianClient, mempoolService, nodes, workDir, serviceConfiguration);
 			wallet.NewFilterProcessed += Wallet_NewFilterProcessed;
 
 			var scp = new Key().ScriptPubKey;
-			var validOperationList = new[] { new WalletService.Operation(scp, Money.Coins(1), "") };
-			var invalidOperationList = new[]
-			{
-				new WalletService.Operation(scp, Money.Coins(10 * 1000 * 1000), ""),
-				new WalletService.Operation(scp, Money.Coins(12 * 1000 * 1000), "")
-			};
-			var overflowOperationList = new[]
-			{
-				new WalletService.Operation(scp, Money.Satoshis(long.MaxValue), ""),
-				new WalletService.Operation(scp, Money.Satoshis(long.MaxValue), ""),
-				new WalletService.Operation(scp, Money.Satoshis(5), "")
-			};
+
+			var validIntent = new PaymentIntent(scp, Money.Coins(1));
+			var invalidIntent = new PaymentIntent(
+				new DestinationRequest(scp, Money.Coins(10 * 1000 * 1000)),
+				new DestinationRequest(scp, Money.Coins(12 * 1000 * 1000)));
+
+			Assert.Throws<OverflowException>(() => new PaymentIntent(
+				new DestinationRequest(scp, Money.Satoshis(long.MaxValue)),
+				new DestinationRequest(scp, Money.Satoshis(long.MaxValue)),
+				new DestinationRequest(scp, Money.Satoshis(5))));
 
 			Logger.TurnOff();
-			// toSend cannot be null
-			Assert.Throws<ArgumentNullException>(() => wallet.BuildTransaction(null, null, 0));
+			Assert.Throws<ArgumentNullException>(() => wallet.BuildTransaction(null, null, FeeStrategy.CreateFromConfirmationTarget(4)));
 
 			// toSend cannot have a null element
-			Assert.Throws<ArgumentNullException>(() => wallet.BuildTransaction(null, new[] { (WalletService.Operation)null }, 0));
+			Assert.Throws<ArgumentNullException>(() => wallet.BuildTransaction(null, new PaymentIntent(new[] { (DestinationRequest)null }), FeeStrategy.CreateFromConfirmationTarget(0)));
 
 			// toSend cannot have a zero element
-			Assert.Throws<ArgumentException>(() => wallet.BuildTransaction(null, new WalletService.Operation[0], 0));
+			Assert.Throws<ArgumentException>(() => wallet.BuildTransaction(null, new PaymentIntent(new DestinationRequest[] { }), FeeStrategy.SevenDaysConfirmationTargetStrategy));
```

Comment:
Why `FeeStrategy.SevenDaysConfirmationTargetStrateg` is used here?

Reply:
@lontivero was expecting the function will fail because `new WalletService.Operation[0]`. But if that would've been correct, it would've failed because of the conf target 0 specified.  

I created the FeeStrategy class and there fee target 0 fails before the expected thingy to fail. We cannot feed the txbuilder without FeeStrategy so I had to fix that first so the expected things will fail first.